src/macros.jl

```diff
@@ -0,0 +1,145 @@
+                # passed as arguments can also be regular expressions, so we just use the
+                # type `Any` for `ignore_content`.
+                ignore_content = Any["[ Info: You just called `trixi_include`. Julia may now compile the code, please be patient.\n"]
+                append!(ignore_content, $additional_ignore_content)
+                for pattern in ignore_content
+                    stderr_content = replace(stderr_content, pattern => "")
+                end
+
+                # We also ignore simple module redefinitions for convenience. Thus, we
+                # check whether every line of `stderr_content` is of the form of a
+                # module replacement warning.
+                @test occursin(r"^(WARNING: replacing module .+\.\n)*$", stderr_content)
+                ret
+            finally
+                rm(fname, force = true)
+            end
+        end
+    end
+end
+
+"""
+    @test_trixi_include(elixir; l2=nothing, linf=nothing, RealT=Float64,
+                                atol=500*eps(RealT), rtol=sqrt(eps(RealT)),
+                                parameters...)
+
+Test Trixi by calling `trixi_include(elixir; parameters...)`.
+By default, only the absence of error output is checked.
+If `l2` or `linf` are specified, in addition the resulting L2/Linf errors
+are compared approximately against these reference values, using `atol, rtol`
+as absolute/relative tolerance.
+"""
+macro test_trixi_include(elixir, args...)
+    # Note: The variables below are just Symbols, not actual errors/types
+    local l2 = get_kwarg(args, :l2, nothing)
+    local linf = get_kwarg(args, :linf, nothing)
+    local RealT_symbol = get_kwarg(args, :RealT, :Float64)
+    RealT = getfield(@__MODULE__, RealT_symbol)
+    atol_default = 500 * eps(RealT)
+    rtol_default = sqrt(eps(RealT))
+    local atol = get_kwarg(args, :atol, atol_default)
+    local rtol = get_kwarg(args, :rtol, rtol_default)
+
+    local kwargs = Pair{Symbol, Any}[]
+    for arg in args
+        if (arg.head == :(=) &&
+            !(arg.args[1] in (:l2, :linf, :RealT, :atol, :rtol)))
+            push!(kwargs, Pair(arg.args...))
+        end
+    end
+
+    quote
+        mpi_isroot() && println("═"^100)
+        mpi_isroot() && println($(esc(elixir)))
+
+        # if `maxiters` is set in tests, it is usually set to a small number to
+        # run only a few steps - ignore possible warnings coming from that
+        if any(==(:maxiters) ∘ first, $kwargs)
+            additional_ignore_content = [
+                r"┌ Warning: Interrupted\. Larger maxiters is needed\..*\n└ @ SciMLBase .+\n",
+                r"┌ Warning: Interrupted\. Larger maxiters is needed\..*\n└ @ Trixi .+\n"]
+        else
+            additional_ignore_content = []
+        end
+
+        # evaluate examples in the scope of the module they're called from
+        @trixi_test_nowarn trixi_include(@__MODULE__, $(esc(elixir)); $kwargs...) additional_ignore_content
```

Comment:
I just realize: This could be problematic. Usually, each package needs another modified `@test_nowarn` macro, which excludes different warnings that should be allowed. That's what the `@test_nowarn_mod` macro defined in the different packages is for. I'm not sure yet what the best way would be to allow for that flexibility.

Reply:
I pushed a draft, which seems to work.